src/hotspot/share/jfr/leakprofiler/chains/bitset.inline.hpp

```diff
@@ -68,7 +68,7 @@ inline BitMap::idx_t BitSet::addr_to_bit(uintptr_t addr) const {
 
 inline CHeapBitMap* BitSet::get_fragment_bits(uintptr_t addr) {
   uintptr_t granule = addr >> _bitmap_granularity_shift;
-  if (granule == _last_fragment_granule) {
+  if (granule == _last_fragment_granule && _last_fragment_bits != NULL) {
```

Comment:
Did you consider changing the initial value of _last_fragment_granule to an invalid granule value (say ~uintptr_t(0))?

That would allow you to keep the old code, which I think  is a little bit easier to understand. With that, either _last_fragment_granule is uninitialized and will never match `granule`, or it has a valid granule value, which can be used to check if we have a properly set up _last_fragment_bits.

Reply:
Good idea, thanks! I've updated the PR.